test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,161 @@
+		require.Empty(t, logs, "Logs were found when none were expected.")
+
+		// If logs are found, print their content for debugging
+		if len(logs) != 0 {
+			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log")
+			t.Logf("Logs detected when none were expected: %v", cat)
+			require.Empty(t, logs, "Logs were found when none were expected.")
+		}
+	}, 5*time.Minute, 10*time.Second)
+
+	// Part 2: Adjust permissions of new log file
+	_, err := s.Env().VM.ExecuteWithError("sudo chmod 777 /var/log/hello-world.log")
+	require.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+
+	// Generate log
+	generateLog(s, t, "hello-world")
+
+	// Part 3: Assert that logs are found in intake after generation
+	checkLogs(s, "hello", "hello-world")
+}
+
+func (s *vmFakeintakeSuite) LogPermission() {
+	t := s.T()
+
+	// Part 4: Block permission and check the Agent status
+	s.Env().VM.Execute("sudo chmod 000 /var/log/hello-world.log")
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		// Check the Agent status
+		statusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
+		require.NoError(t, err, "Issue running agent status: %s", err)
+
+		if strings.Contains(statusOutput, "Status: OK") {
+			require.Fail(t, "log file is unexpectedly accessible")
+		}
+
+		require.Contains(t, statusOutput, "denied", "Log file is correctly inaccessible")
+	}, 3*time.Minute, 10*time.Second)
+
+	// Part 5: Restore permissions
+	s.Env().VM.Execute("sudo chmod 777 /var/log/hello-world.log")
+
+	// Part 6: Restart the agent, generate new logs
+	s.Env().VM.Execute("sudo service datadog-agent restart")
+
+	generateLog(s, s.T(), "hello-world")
+
+	// Check the Agent status
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		statusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
+		require.NoError(t, err, "Issue running agent status: %s", err)
+		require.Contains(t, statusOutput, "Status: OK", "Expecting log file to be accessible but it is inaccessible instead")
+	}, 5*time.Minute, 2*time.Second)
+}
+
+func (s *vmFakeintakeSuite) LogRotation() {
+	t := s.T()
+
+	// Part 7: Rotate the log file and check if the agent is tailing the new log file.
+	// Rotate the log file
+	s.Env().VM.Execute("sudo mv /var/log/hello-world.log /var/log/hello-world.log.old && sudo touch /var/log/hello-world.log")
+
+	// Verify the old log file's existence after rotation
+	_, err := s.Env().VM.ExecuteWithError("ls /var/log/hello-world.log.old")
+	require.NoError(t, err, "Failed to find the old log file after rotation")
+
+	// Grant new log file permission
+	s.Env().VM.Execute("sudo chmod 777 /var/log/hello-world.log")
+
+	// Check if agent is tailing new log file via agent status
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		newStatusOutput, err := s.Env().VM.ExecuteWithError("sudo datadog-agent status | grep -A 10 'custom_logs'")
```

Comment:
Is there any built-in support to query the status object before its turned into template text? 
Ie, instead of using `agent status` to query, query the status HTTP endpoint directly and expose the returned JSON?

I'm hesitant to rely too much on `agent status` for tests in general, but it would be less error-prone if we're querying and asserting on a specific json field vs `assert.Contains`

Reply:
We could move [this implementation](https://github.com/DataDog/datadog-agent/blob/0730fb490d11374c8b9158ff9d464c0a6bcd6408/test/new-e2e/tests/agent-platform/common/agent_behaviour.go#L31) by @KevinFairise2 to common helpers, it executes the status command on the agent client (s.Env().Agent) and parses it using json library

Reply:
Agreed you can skip status, except if you want to test status, and check fakeintake directly